From 99db8d62a62610f9aed8225c47dc051b5d61b1a1 Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Wed, 05 Sep 2018 22:36:45 +0000
Subject: [PATCH] Disable cfi-icall for media/gpu/vaapi/va_stubs.cc

Similarly to pulse_stubs.cc, va_stubs.cc contains dlsym() resolved
functions which trigger cfi-icall failures. The affected builds of
Chromium do use out-of-tree patches to enable hardware-accelerated
video decode on Linux where the VA-API driver is installed, but it
would still be nice to have them work with cfi-icall enabled.

Change-Id: Iefa27f4b95007da23423d17727826adb8825b278
Reviewed-on: https://chromium-review.googlesource.com/1208830
Reviewed-by: Peter Collingbourne <pcc@chromium.org>
Commit-Queue: Peter Collingbourne <pcc@chromium.org>
Cr-Commit-Position: refs/heads/master@{#589034}
---

--- a/tools/cfi/blacklist.txt
+++ b/tools/cfi/blacklist.txt
@@ -143,6 +143,7 @@ fun:*FunctorTraits*
 
 # Calls to auto-generated stubs by generate_stubs.py
 src:*audio/pulse/pulse_stubs.cc
+src:*media/gpu/vaapi/va_stubs.cc
 
 # Calls to auto-generated stubs by generate_library_loader.py
 src:*chrome/browser/speech/tts_linux.cc
